docs/development/test-vectors.rst

```diff
@@ -546,6 +546,9 @@ Custom X.509 Vectors
   This is an invalid certificate per :rfc:`5280` 4.2.1.12.
 * ``malformed-san.pem`` - A certificate with a malformed SAN.
 * ``malformed-ian.pem`` - A certificate with a malformed IAN.
+* ``pkcs7/ca/rsa_ca.pem`` - A self-signed RSA certificate with ``basicConstraints``
```

Comment:
I don't understand what this is going to be used for. I can't really parse "This key is encoded is used to attempt decoding the PKCS7 custom vectors", but if it's just being used to test failed decrypts is there any reason we need a new vector here?

Reply:
That's a very good point actually! Two thoughts there:
- Is there any reason to use a RSA certificate from x509/custom/ca? Maybe a separate PR to use a dedicated CA chain for PKCS7 tests can be interesting?
- For now, if you point me towards an existing RSA cert chain, I can definitely use it for testing errors in PKCS7 decryption. I haven't found an appropriate one (probably haven't searched enough).

Reply:
I'm fine with a dedicated CA chain, although please make sure it's got enough lifetime in it that either quantum computers come into being or 20 years from now we upgrade it out of guilt 😄

If you need to build a valid chain of length > 1 then we typically just dynamically build those off a root (see `test_x509.py`).

Reply:
For now, for this type of testing (no_recipient_match) I'd rather dynamically create a chain. I'll look into `test_x509.py` and let you know. If I can manage to do without a dedicated chain, I'll update #11555 and this pull request accordingly :)

Reply:
I've managed to do something similar to `test_x509.py`, so I'm now only adding one new vector.